NickvisionTubeConverter.Shared/Models/Timeframe.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Globalization;
+
+namespace NickvisionTubeConverter.Shared.Models;
+
+/// <summary>
+/// A model of a timeframe
+/// </summary>
+public class Timeframe
+{
+    /// <summary>
+    /// The start of the timeframe
+    /// </summary>
+    public TimeSpan Start { get; init; }
+    /// <summary>
+    /// The end of the timeframe
+    /// </summary>
+    public TimeSpan End { get; init; }
+    
+    /// <summary>
+    /// Constructs a timeframe
+    /// </summary>
+    /// <param name="startTime">The start TimeSpan</param>
+    /// <param name="endTime">The end TimeSpan</param>
+    public Timeframe(TimeSpan start, TimeSpan end)
+    {
+        Start = start;
+        End = end;
+    }
+    
+    /// <summary>
+    /// Parses a Timeframe object
+    /// </summary>
+    /// <param name="start">The start time</param>
+    /// <param name="start">The end time</param>
+    /// <returns>The Timeframe object</returns>
+    public static Timeframe Parse(string start, string end, double duration)
+    {
+        TimeSpan startSpan;
+        TimeSpan endSpan;
+        try
+        {
+            startSpan = TimeSpan.Parse(start, CultureInfo.CurrentCulture);
+        }
+        catch
+        {
+            throw new ArgumentException("Unable to parse start time.");
+        }
+        try
+        {
+            endSpan = TimeSpan.Parse(end, CultureInfo.CurrentCulture);
+        }
+        catch
+        {
+            throw new ArgumentException("Unable to parse emd time.");
+        }
+        if(startSpan < TimeSpan.FromSeconds(0))
+        {
+            throw new ArgumentException("Invalid start time.");
+        }
+        if(endSpan < startSpan + TimeSpan.FromSeconds(1) || endSpan > TimeSpan.FromSeconds(duration))
```

Comment:
Why +1 second?

Reply:
That's a question for @fsobolev,  this was his code from an old branch we had.

Reply:
@DaPigGuy Ah i see why he did that.

Let say you have a start time of `00:00:00`. Without adding `TimeSpan.FromSeconds(1)`, `00:00:00` will be considered a valid end time. But it's not because your downloading nothing. Therefore `00:00:01` and above will be considered valid.

Reply:
Wouldn't a `<=` comparison be more appropriate